src/StateManager.ts

```diff
@@ -117,6 +118,37 @@ export default class StateManager {
     return StateManager.clean;
   }
 
+  /** Marks the file as clean. */
+  static listeners = new Set<() => void>();
+
+  static setDebug() {
+    if (this.debugActive) {
+      StateManager.notifyListeners();
+    }
+
+    StateManager.debugActive = !StateManager.debugActive;
+  }
+
+  static notifyListeners() {
+    Array.from(StateManager.listeners).forEach((listener) => listener());
+  }
+
+  static subscribe(listener: () => void) {
+    StateManager.listeners.add(listener);
+  }
+
+  static unsubscribe(listener: () => void) {
+    StateManager.listeners.delete(listener);
+  }
+
+  /** Returns the the curent clean state of the file */
+
+  public static checkDebug() {
+    return StateManager.debugActive;
+  }
+  public static get debuggEnabled(): boolean {
```

Comment:
```suggestion
  public static get debugEnabled(): boolean {
```
"debug" is misspelled here